server.js

```diff
@@ -2,22 +2,137 @@ const express = require("express");
 const cors = require("cors");
 
 const app = express();
+app.use(express.json())
 
 app.use(cors());
 
-const welcomeMessage = {
-  id: 0,
-  from: "Bart",
-  text: "Welcome to CYF chat system!",
-};
+const messages = [
+  {
+    id: 0,
+    from: "Bart",
+    text: "Welcome to CYF chat system!",
+    
+  }]
 
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+// const messages = [welcomeMessage];
 
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
+//////////////////////Create a new message//////////////////////////// 
+ app.post('/messages/', (req, res) =>{
+  let newId = messages[messages.length -1].id + 1
+  let newMessage = {
+        id: newId,
+        from: req.body.from,
+        text: req.body.text,
+        created_at :new Date().toISOString()
+      };
+ 
+  if((newMessage.from !== undefined) && (newMessage.from.length > 0) && (newMessage.text !== undefined)&& (newMessage.from.length > 0) ){
```

Comment:
Exhaustive validation! I would also suggest factoring it out to a separate function to decouple the logic and make it easier to read the code.
Also, you can use the so-called "[chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)" to simplify undefined checking e.g
`(newMessage?.from.length > 0)`
It'd halve your condition.